source/dub/dub.d

```diff
@@ -546,10 +546,11 @@ class Dub {
 			BuildSettingsTemplate tcinfo = m_project.rootPackage.recipe.getConfiguration(config).buildSettings;
 			tcinfo.targetType = TargetType.executable;
 			tcinfo.targetName = test_config;
-			// HACK for vibe.d's legacy main() behavior:
-			tcinfo.versions[""] ~= "VibeCustomMain";
-			m_project.rootPackage.recipe.buildSettings.versions[""] = m_project.rootPackage.recipe.buildSettings.versions.get("", null).remove!(v => v == "VibeDefaultMain");
-			// TODO: remove this ^ once vibe.d has removed the default main implementation
+
+			if (lbuildsettings.targetType != TargetType.executable) {
+				logInfo("target type is %s, which doesn't have a main(); we're providing one");
```

Comment:
Not sure whether the user usually needs to care about this. I'd change this to `logDiagnostic` to avoid polluting the output unnecessarily.